### DISCUSSION TITLE
Create-next-app doesn't allow a '.' as a project name

### DISCUSSION THREAD

**Discussion by LizAinslie** [vercel/next.js#62630] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

not relevant, issue is with create-next-app

### Describe the Bug

When I run `yarn create next-app --typescript` in the JetBrains terminal window, and try to enter `.` as a project name, I get an error, `Invalid project name: name can no longer contain capital letters`.

Here's a screenshot: ![](https://i-work-at-the.cocaine.institute/Lizzy630f6b03AByOFFQ7hoo7.png)

If the `.` is not allowed, I think it makes sense to replace the error, but I think there ought to be a way to put a project in my current directory so I don't have to go manually move everything later.

### Expected Behavior

I expect to be able to create a project in the current directory using `yarn create next-app`

### Link to reproduction

  how do I reproduce something in create-next-app?

### To Reproduce

![](https://i-work-at-the.cocaine.institute/Lizzy630f6b03AByOFFQ7hoo7.png) Ignore the cr, that was a typo and a terminal artifact duplicated the line

**Comment by wwwdepot** [comment]
create-next-app uses `validate-npm-package-name` to validate project name. Your current directory is not a valid npm package name, use all lowercase letters, and try again.

https://github.com/vercel/next.js/blob/ef69aca4df648949aa0011b1431fafd6f21dafe5/packages/create-next-app/index.ts#L77

**Comment by LizAinslie** [comment]
This still seems untoward and limiting. It's not a huge deal but it's a QoL feature that'd be really nice to have if I could opt to initialize a project in the current directory. Even a question defaulting to "No" would be better IMO

**Comment by samcx** [comment]
Moving this to discussions since this is technically a feature request!

**Comment by Wxh16144** [comment]
Are there any updates? I'm trying to learn about a certain topic, and usually, I would first create a folder and then initialize the project. For example, recently, I wanted to look at something related to RAG, but I got a warning when I created the project...

By the way, the vite cli next door is really very useful.
<img width="816" height="170" alt="image" src="https://github.com/user-attachments/assets/5f7677ee-a9d7-44de-84d8-b235d811c6aa" />
